src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -168,6 +168,14 @@ private async Task ExecuteAsync(HttpContext context, ConnectionDelegate connecti
             {
                 transport = HttpTransportType.WebSockets;
                 connection = await GetOrCreateConnectionAsync(context, options);
+
+                if (connection is not null)
```

Comment:
how can this be null?

Reply:
If you attempt to connect with a connection id query string param that doesn't exist on this server instance.

Reply:
This method name is bad then, GetOrCreateOrNot

Reply:
😆 this method name is >8 years old now.

Reply:
I blame myself

Reply:
I do too 😉